ui/src/AuthService.js

```diff
@@ -0,0 +1,55 @@
+import config from './config/config';
+
+/**
+ * Creates a configuration obect with information about the request type
+ * and the request body
+ *
+ * @param {string} method - HTTP Method (PUT, GET, PATCH, DELETE, POST)
+ * @param {object|Array} body - JSON Payload to send in the request
+ * @returns a config object to send in HTTP requests
+ */
+const requestConfig = (method, body) => ({
+  method,
+  cache: 'no-cache',
+  credentials: 'same-origin',
+  headers: {
+    'Content-Type': 'application/json; charset=utf-8'
+  },
+  body: body && JSON.stringify(body)
+});
+
+/**
+ * Sends a HTTP request
+ *
+ * @param {string} path - The URI to send the request to
+ * @param {*} method - HTTP Method (PUT, GET, PATCH, DELETE, POST)
+ * @param {*} body - The request payload
+ * @returns {boolean|JSON} - Returns the response from the HTTP request
+ */
+const request = async (path, method, body) => {
+  const response = await fetch(`${config.auth.url}/${path}`, requestConfig(method, body));
+  if (response.status !== 202) {
+    console.log(`${config.auth.url}/${path}`);
+    const msg = await response.text();
+
+    throw Error(msg);
+  }
+
+  return response;
+};
+
+/**
+ * Makes a HTTP GET request to a provided URI and returns a single item
+ *
+ * @param {string} url - The full URL to make the GET request to
+ */
+const fetchItem = async url => {
+  const result = await request(url, 'GET');
```

Comment:
The value passed here is url but it's accepted as path in the async method. That method also has a url value which is coming from the the config. It could be bit clearer to call url path here?

Reply:
That's true, I just copied what's in the DataService. I will update it.